index.ts

```diff
@@ -1,35 +1,65 @@
 export class OatyArray {
-  private _original: object[]
-  private _keys: string[]
-  private _transposed: object = {}
+  private keys: string[] | undefined
+  private original: object[] = []
+  private transposed: { [key: string]: any } = {}
+
+  constructor(options?: { data?: object[], keys?: string[] }) {
+    if (options !== undefined) {
+      if (options.data !== undefined) { this.push(...options.data) }
+      if (options.keys !== undefined) { this.keys = options.keys }
+    }
+  }
 
   get length(): number {
-    return this._original.length
+    return this.original.length
   }
 
-  constructor(input: object[], keys: string[]) {
-    this._original = input.map((x) => Object.assign({}, x))
-    this._keys = keys
+  public getKeys(): string[] | undefined {
+    return this.keys
+  }
 
-    this._transposed = transpose(this._keys, this._original)
+  public getOriginal(): object[] {

```

Comment:
should be a getter function - see above comment

Reply:
Resolved